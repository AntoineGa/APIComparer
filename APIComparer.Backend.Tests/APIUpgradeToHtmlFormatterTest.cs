namespace APIComparer.Tests.VersionComparisons
{
    using System.IO;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using APIComparer.Backend;
    using APIComparer.Backend.Reporting;
    using APIComparer.VersionComparisons;
    using NUnit.Framework;

    [TestFixture]
    [UseReporter(typeof(DiffReporter),typeof(ClipboardReporter))]
    public class APIUpgradeToHtmlFormatterTest
    {
        [Test]
        public void TestComplianceNewtonsoftJson()
        {
            Approvals.VerifyHtml(HtmlDiff("newtonsoft.json", "5.0.8", "6.0.8"));
        }

        [Test]
        public void TestComplianceNServiceBus()
        {
            Approvals.VerifyHtml(HtmlDiff("nservicebus", "4.0.0", "5.0.0"));
        }

        [Test]
        public void TestComplianceAzureSelfDestruct()
        {
            Approvals.VerifyHtml(HtmlDiff("Two10.Azure.SelfDestruct", "1.0.0", "1.0.5"));
        }

        [Test]
        public void TestComplianceLibLog()
        {
            Approvals.VerifyHtml(HtmlDiff("LibLog", "3.0.0", "4.1.1"));
        }

        [Test]
        public void AppccelerateEventBrokerNoChanges()
        {
            Approvals.VerifyHtml(HtmlDiff("Appccelerate.EventBroker", "3.1.0", "3.15.0"));
        }

        private string HtmlDiff(string packageName, string fromVersion, string untilVersion)
        {
            using (var writer = new StringWriter())
            {
                var formatter = new APIUpgradeToHtmlFormatter();

                var packageDescription = new PackageDescription
                {
                    PackageId = packageName,
                    Versions = new VersionPair(fromVersion, untilVersion)
                };

                var compareSetCreator = new CompareSetCreator();
                var sets = compareSetCreator.Create(packageDescription);
                var compareSetDiffer = new CompareSetDiffer();
                var diff = compareSetDiffer.Diff(sets);

                formatter.Render(writer, packageDescription, diff);

                return writer.ToString();
            }
        }
    }
}
