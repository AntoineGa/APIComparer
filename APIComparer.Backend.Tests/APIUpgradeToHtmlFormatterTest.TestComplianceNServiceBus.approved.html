
<head><link rel="stylesheet" type="text/css" href="/Content/style.css" ></head>
<body>
<section id=".NETFramework,Version=v4.5" title=".NETFramework,Version=v4.5">
<a class="tab" href="#.NETFramework,Version=v4.5">.NETFramework,Version=v4.5</a>
<div class="content">
<h1>.NETFramework,Version=v4.5</h1>
        
        
        
<h2>The following types have differences.</h2>

  <h3>NServiceBus.ConfigureFileShareDataBus</h3>











<h4>Methods Obsoleted</h4>
<ul>
  <li><code>void FileShareDataBus(NServiceBus.BusConfiguration, string)</code><br/> Use `configuration.UseDataBus<FileShareDataBus>().BasePath(basePath)`, where <code>configuration</code> is an instance of <code>BusConfiguration</code>. If self-hosting the instance can be obtained from `new BusConfiguration()`. if using the NServiceBus Host the instance of <code>BusConfiguration</code> will be passed in via the <code>INeedInitialization</code> or <code>IConfigureThisEndpoint</code> interfaces. Will be treated as an error from version 5.5.0. Will be removed in version 6.0.0. Obsoleted with warning.</li>
</ul>


  <h3>NServiceBus.Sagas.ActiveSagaInstance</h3>











<h4>Methods Obsoleted</h4>
<ul>
  <li><code>NServiceBus.Saga.Saga get_Instance()</code><br/> Please use <code>context.MessageHandler.Instance</code> instead. Will be treated as an error from version 6.0.0. Will be removed in version 7.0.0. Obsoleted with warning.</li>
  <li><code>Type get_SagaType()</code><br/> Please use <code>.Metadata.SagaType</code> instead. Will be treated as an error from version 6.0.0. Will be removed in version 7.0.0. Obsoleted with warning.</li>
</ul>


    <h3>NServiceBus.Unicast.IMessageHandlerRegistry - Obsoleted</h3>
    <p>Not a public API. Please use `MessageHandlerRegistry` instead. Will be treated as an error from version 6.0.0. Will be removed in version 6.0.0. Obsoleted with warning.</p>














  <h3>NServiceBus.Unicast.UnicastBus</h3>











<h4>Methods Obsoleted</h4>
<ul>
  <li><code>NServiceBus.Hosting.HostInformation get_HostInformation()</code><br/> We have introduced a more explicit API to set the host identifier, see busConfiguration.UniquelyIdentifyRunningInstance(). Will be treated as an error from version 6.0.0. Will be removed in version 7.0.0. Obsoleted with warning.</li>
  <li><code>void set_HostInformation(NServiceBus.Hosting.HostInformation)</code><br/> We have introduced a more explicit API to set the host identifier, see busConfiguration.UniquelyIdentifyRunningInstance(). Will be treated as an error from version 6.0.0. Will be removed in version 7.0.0. Obsoleted with warning.</li>
</ul>


  <h3>NServiceBus.WireEncryptedString</h3>











<h4>Methods Obsoleted</h4>
<ul>
  <li><code>string get_Base64Iv()</code><br/> No longer required. Will be treated as an error from version 6.0.0. Will be removed in version 6.0.0. Obsoleted with warning.</li>
  <li><code>string get_EncryptedBase64Value()</code><br/> No longer required. Will be treated as an error from version 6.0.0. Will be removed in version 6.0.0. Obsoleted with warning.</li>
  <li><code>void set_Base64Iv(string)</code><br/> No longer required. Will be treated as an error from version 6.0.0. Will be removed in version 6.0.0. Obsoleted with warning.</li>
  <li><code>void set_EncryptedBase64Value(string)</code><br/> No longer required. Will be treated as an error from version 6.0.0. Will be removed in version 6.0.0. Obsoleted with warning.</li>
</ul>


    </div>
</section>
</body>